backend/app/views.py

```diff
@@ -111,6 +113,39 @@ def get_anagram(request, text_id, part_of_speech):
     }
     return Response(res)
 
+
+@api_view(['GET'])
+def get_picturebook_prompt(request, text_id, part_of_speech):
+    """
+    API endpoint for getting the necessary information for the picture book exercise
+    given the id of the text and the part of speech. The words chosen will be random.
+    """
+    text_obj = Text.objects.get(id=text_id)
+    image_urls = text_obj.images
+    words = get_valid_words(text_obj.content, part_of_speech)
+    random.shuffle(words)
+    words = words[:4]
+
+    res = [{'word': word,
+            'url': image_urls.get(word, '')}
+           for word in words]
+    return Response(res)
+
+
+@api_view(['GET'])
+def get_picturebook_data(request):
+    """
+    API endpoint for getting the picture book images given the story the user wrote.
+    """
+    urls = get_story_data(request.GET.get('content'))
+    misspelled = get_misspelled_words(request.GET.get('content'))
```

Comment:
One comment for you @joshfeli: upon further reading, it looks like the [Django REST framework docs](https://www.django-rest-framework.org/api-guide/requests/#query_params) recommend the use of `request.query_params` instead of `request.GET`. They are synonymous, but `query_params` is more descriptive. It's probably worth updating this here to set our standards for the future.